test/Window/WindowHandleWin32.test.cpp

```diff
@@ -36,26 +37,84 @@ LRESULT WINAPI wndProc(HWND handle, UINT message, WPARAM wParam, LPARAM lParam)
     }
     return DefWindowProcW(handle, message, wParam, lParam);
 }
+
+struct NormalWindow
+{
+};
+struct NormalMenuWindow
+{
+};
+struct ExStyleWindow
+{
+};
+struct ExStyleMenuWindow
+{
+};
+
+template <typename WindowType>
+HWND createWindow(LPWSTR className, HINSTANCE hInstance);
+
+HWND createWindowWrapper(LPWSTR className, HINSTANCE hInstance, DWORD dwExStyle, HMENU hMenu)
+{
+    return CreateWindowExW(dwExStyle,
+                           className,
+                           L"WindowHandle Tests",
+                           WS_OVERLAPPEDWINDOW,
+                           CW_USEDEFAULT,
+                           CW_USEDEFAULT,
+                           640,
+                           480,
+                           nullptr,
+                           hMenu,
+                           hInstance,
+                           nullptr);
+}
+
+
+template <>
+HWND createWindow<NormalWindow>(LPWSTR className, HINSTANCE hInstance)
+{
+    return createWindowWrapper(className, hInstance, 0, nullptr);
+}
+
+template <>
+HWND createWindow<NormalMenuWindow>(LPWSTR className, HINSTANCE hInstance)
+{
+    HMENU menu = CreateMenu();
+    AppendMenuW(menu, MF_STRING, 1, L"Test");
+    return createWindowWrapper(className, hInstance, 0, menu);
+}
+
+template <>
+HWND createWindow<ExStyleWindow>(LPWSTR className, HINSTANCE hInstance)
+{
+    return createWindowWrapper(className, hInstance, WS_EX_TOOLWINDOW | WS_EX_CLIENTEDGE, nullptr);
+}
+
+template <>
+HWND createWindow<ExStyleMenuWindow>(LPWSTR className, HINSTANCE hInstance)
+{
+    HMENU menu = CreateMenu();
+    AppendMenuW(menu, MF_STRING, 1, L"Test");
+    return createWindowWrapper(className, hInstance, WS_EX_TOOLWINDOW | WS_EX_CLIENTEDGE, menu);
+}
+
 } // namespace
 
-TEST_CASE("[Window] sf::WindowHandle (Win32)")
+TEMPLATE_TEST_CASE("[Window] sf::WindowHandle (Win32)",
+                   "[!mayfail][!nonportable]",
+                   NormalWindow,
+                   NormalMenuWindow,
+                   ExStyleWindow,
+                   ExStyleMenuWindow)
```

Comment:
I tried passing an enum instead of the types here, but that didn't really compile so now I just used these dummy structs.